presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -235,7 +269,8 @@ private void schedule()
                     break;
                 }
 
-                List<SectionExecution> sectionExecutions = createStageExecutions(sectionsReadyForExecution);
+                // Apply runtime CBO on the ready sections before creating SectionExecutions.
+                List<SectionExecution> sectionExecutions = createStageExecutions(sectionsReadyForExecution.stream().map(this::tryCostBasedOptimize).collect(toImmutableList()));
```

Comment:
Let's put `.map(this::tryCostBasedOptimize)` and `.collect(toImmutableList()))` on the new line each